sci-misc/boinc/files/boinc.init

```diff
@@ -23,30 +23,45 @@ create_work_directory() {
 		chown "${USER}:${GROUP}" "${RUNTIMEDIR}"
 	fi
 
-	if [[ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]]; then
+	if [ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]; then
 		ln -s /etc/ssl/certs/ca-certificates.crt "${RUNTIMEDIR}"/ca-bundle.crt
 	fi
 
 	return 0
 }
 
 
+fix_lib_symlinks() {
+	local src="$1"
+	local tgt="$2"
+
+	if test -n "${tgt}"; then
+		# Remove a broken symlink
+		if test -h "${tgt}" \
+			-a "${src}" != "$(readlink "${tgt}")" ; then
+			rm -f "${tgt}"
+		fi
+
+		# symlink the correct path
+		if test -n "${src}" -a -f "${src}" -a ! -h "${tgt}"; then
+			ln -snf "$src" "$tgt"
+		fi
+	fi
+}
+
 cuda_check() {
-	local libtarget="${RUNTIMEDIR}/libcudart.so"
 	local libsource="$(ls -t /opt/cuda/lib*/libcudart.so 2>/dev/null | head -n 1)"
+	local libtarget="${RUNTIMEDIR}/libcudart.so"
 
-	# Remove a broken symlink
-	if [[ -h "${libtarget}" ]] \
-	&& [[ "${libsource}" != "$(readlink "${libtarget}")" ]]; then
-		rm -f "${libtarget}"
-	fi
+	fix_lib_symlinks "$libsource" "$libtarget"
+}
 
-	# symlink the correct path
-	if [[ -n "${libsource}" ]] \
-	&& [[ -f "${libsource}" ]] \
-	&& [[ ! -h "${libtarget}" ]]; then
-		ln -snf "$libsource" "${libtarget}"
-	fi
+
+opencl_check() {
+	local libsource="$(ls /usr/lib/libOpenCL.so 2>/dev/null)"
```

Comment:
Likewise. Plus this won't work on modern amd64 where `/usr/lib` is not a symlink.